database/engine/metadata_log/metalogpluginsd.c

```diff
@@ -116,41 +125,55 @@ PARSER_RC metalog_pluginsd_context_action(void *user, uuid_t *uuid)
 
     ret = find_object_by_guid(uuid, object, 49);
     switch (ret) {
-    case GUID_TYPE_CHAR:
-        fatal_assert(0);
-        break;
-    case GUID_TYPE_CHART:
-    case GUID_TYPE_DIMENSION:
-        host = ctx->rrdeng_ctx->host;
-        switch (ret) {
-        case GUID_TYPE_CHART:
-            chart_char_guid = (uuid_t *)(object + 16);
-
-            ret = find_object_by_guid(chart_char_guid, id_str, RRD_ID_LENGTH_MAX + 1);
-            fatal_assert(GUID_TYPE_CHAR == ret);
-            ((PARSER_USER_OBJECT *) user)->st = rrdset_find(host, id_str);
+        case GUID_TYPE_NOTFOUND:
+            if (unlikely(ctx->rrdeng_ctx->host && uuid_compare(ctx->rrdeng_ctx->host->host_uuid, *uuid)))
+                error_with_guid(uuid, "Failed to find valid context");
+            break;
+        case GUID_TYPE_CHAR:
+            error_with_guid(uuid, "Ignoring unexpected type GUID_TYPE_CHAR");
             break;
+        case GUID_TYPE_CHART:
         case GUID_TYPE_DIMENSION:
-            chart_guid = (uuid_t *)(object + 16);
-
-            ret = find_object_by_guid(chart_guid, chart_object, 33);
-            fatal_assert(GUID_TYPE_CHART == ret);
-            chart_char_guid = (uuid_t *)(chart_object + 16);
-
-            ret = find_object_by_guid(chart_char_guid, id_str, RRD_ID_LENGTH_MAX + 1);
-            fatal_assert(GUID_TYPE_CHAR == ret);
-            ((PARSER_USER_OBJECT *) user)->st = rrdset_find(host, id_str);
+            host = ctx->rrdeng_ctx->host;
+            switch (ret) {
+                case GUID_TYPE_CHART:
+                    chart_char_guid = (uuid_t *)(object + 16);
+
+                    ret = find_object_by_guid(chart_char_guid, id_str, RRD_ID_LENGTH_MAX + 1);
+                    if (unlikely(GUID_TYPE_CHAR != ret))
+                        error_with_guid(uuid, "Failed to find valid chart name");
+                    else
+                        ((PARSER_USER_OBJECT *)user)->st = rrdset_find(host, id_str);
+                    break;
+                case GUID_TYPE_DIMENSION:
```

Comment:
This seems wrong - if the GUID in the context command was a dimension then why treat it as a chart?

Reply:
We don't have a dimension state yet, in theory you could add dimension labels in this context and set `rd=dimension`, `st=dimension's chart`, `host=chart's host`.

Since the only state we have right now is `st`, we just set this.